### PR TITLE
Inject OCSClient - Fixes master unit tests

### LINKED ISSUE #21451: Master unit tests are broken
Ref https://github.com/owncloud/core/pull/21450:

```
15:06:59 There were 4 errors:
15:06:59 
15:06:59 1) OCA\Provisioning_API\Tests\AppsTest::testGetApps
15:06:59 Invalid argument supplied for foreach()
15:06:59 
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/ocsclient.php:155
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:883
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:828
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/lib/apps.php:48
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/tests/appstest.php:79
15:06:59 
15:06:59 2) OCA\Provisioning_API\Tests\AppsTest::testGetAppsEnabled
15:06:59 Invalid argument supplied for foreach()
15:06:59 
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/ocsclient.php:155
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:883
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:828
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/lib/apps.php:48
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/tests/appstest.php:90
15:06:59 
15:06:59 3) OCA\Provisioning_API\Tests\AppsTest::testGetAppsDisabled
15:06:59 Invalid argument supplied for foreach()
15:06:59 
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/ocsclient.php:155
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:883
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:828
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/lib/apps.php:48
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/tests/appstest.php:100
15:06:59 
15:06:59 4) OCA\Provisioning_API\Tests\AppsTest::testGetAppsInvalidFilter
15:06:59 Invalid argument supplied for foreach()
15:06:59 
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/ocsclient.php:155
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:883
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/lib/private/app.php:828
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/lib/apps.php:48
15:06:59 /ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/apps/provisioning_api/tests/appstest.php:114
```

### PR DESCRIPTION
**Pull request by LukasReschke**
Fixes https://github.com/owncloud/core/issues/21451

### CODE REVIEW

File: settings/controller/appsettingscontroller.php [hunk 1]
```diff
@@ -310,7 +310,7 @@ public function listApps($category = '', $includeUpdateInfo = true) {
 	 * @return array
 	 */
 	private function getInstalledApps($includeUpdateInfo = true) {
-		$apps = \OC_App::listAllApps(true, $includeUpdateInfo);
+		$apps = \OC_App::listAllApps(true, $includeUpdateInfo, \OC::$server->getOcsClient());
```
Comment (DeepDiver1975) [inline settings/controller/appsettingscontroller.php]: inject the client into the controller?

Comment (LukasReschke) [inline settings/controller/appsettingscontroller.php]: True.

Comment (LukasReschke) [inline settings/controller/appsettingscontroller.php]: + Done

**Comment by rullzer** [comment]
mmm stupid app store...

:+1: nothing app related seems to break here...

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**